include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -32,10 +39,11 @@ namespace seqan3::detail
  * \tparam cursor_t      Must model seqan3::FmIndexCursor.
  * \tparam query_t       Must model std::ranges::InputRange over the index's alphabet.
  * \tparam delegate_t    Takes `index::cursor_type` as argument.
- * \param[in] cur        Cursor of atring index built on the text that will be searched.
+ * \param[in] cur        Cursor of a string index built on the text that will be searched.
  * \param[in] query      Query sequence to be searched with the cursor.
  * \param[in] query_pos  Position in the query sequence indicating the prefix that has already been searched.
  * \param[in] error_left Number of errors left for matching the remaining suffix of the query sequence.
+ * \param[in] prev_enum  Previous scenario of search(), i.e. error or match.
```

Comment:
```suggestion
 * \param[in] prev_enum  Previous scenario of search, i.e. error or match.
```
It's not the `search()` function to be exact. I think `search` is enough